test-galasactl-ecosystem.sh

```diff
@@ -189,6 +189,151 @@ function launch_test_on_ecosystem_with_portfolio {
+    --portfolio portfolio.yaml \
+    --throttle 1 \
+    --poll 10 \
+    --progress 1 \
+    --noexitcodeontestfailures \
+    --log ${log_file}"
+
+    set -o pipefail # Fail everything if anything in the pipeline fails. Else we are just checking the 'tee' return code.
+
+    
+    $cmd &
+
+    is_done="false"
+    retries=0
+    max=100
+    target_line=""
+    
+    
+    while [[ "${is_done}" == "false" ]]; do
+        if [[ -e $log_file ]]; then
+            echo "file exists"
+            target_line=$(cat ${log_file} | grep "submitted")
+            
+
+            if [[ "$target_line" != "" ]]; then
+                echo "Target line is found."
+                is_done="true"
+            fi
+        fi    
+        sleep 1
+        ((retries++))
+        if (( $retries > $max )); then 
+            echo "Too many retries."
+            is_done="true"
+        fi
+    done
+
+    run_name=$(echo $target_line | cut -f4 -d' ')
+    echo "Run name is $run_name"
+
+    rc=$?
+
+    if [[ "${rc}" != "0" ]]; then 
+        error "Failed to submit a test to a remote ecosystem."
+        exit 1
+    fi
+
+    cmd="${BASEDIR}/bin/${binary} runs download \
+    --name ${run_name} \
+    --bootstrap ${bootstrap} \
+    --force"
+
+    info "Command is: $cmd"
+
+    output_file="runs-download-output.txt"
+
+    is_test_finished="false"
+    retries=0
+    max=100
+    target_line=""
+    while [[ "${is_test_finished}" == "false" ]]; do
+        $cmd | tee $output_file
+      
+        target_line=$(cat ${log_file} | grep "has finished")
+       
+        if [[ "$target_line" != "" ]]; then
+            echo "Target line is found."
+            is_test_finished="true"
+           
+            folder_name=$(cat $output_file| cut -d' ' -f 7)
+            echo $folder_name | grep ":" 
+            rc=$?
+            if [[ "${rc}" != "1" ]]; then 
+                error "Folder named incorrectly. Has timestamp when it should not."
+                exit 1
+            fi
+        sleep 2    
+        else
+            
+            test_building_line=$(cat ${log_file} | grep "now 'building'")
+            if [[ "$test_building_line" != "" ]]; then
+                cat ${log_file} | grep "now 'running'" -q
+                rc=$?
+                if [[ "${rc}" != "0" ]]; then
+        
+                    folder_name=$(cat $output_file| cut -d' ' -f 7)
+                    echo $folder_name | grep ":" 
+                    rc=$?
+                    if [[ "${rc}" != "0" ]]; then 
+                        error "Folder named incorrectly. Has no timestamp when it should."
+                        exit 1
+                    fi
+                fi    
+            fi
+        fi
+
+        
+        sleep 2
+        
+        ((retries++))
+        if (( $retries > $max )); then 
+            echo "Too many retries."
+            is_test_finished="true"
+        fi
+    done
+
+    success "Downloading artifacts from a running test worked OK"
```

Comment:
success "Downloading artifacts from a running test results in folder names with a timestamp. OK"